src/appleseed.shaders/src/appleseed/as_cdl.osl

```diff
@@ -0,0 +1,109 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// American Society of Cinematographers Color Decision List implementation
+
+shader as_cdl
+[[
+    string help = "Slope/Offset/Power Color Decision List modifier according to ASC",
+    string icon = "asCDL.png",
+    string URL = "please fill in",
+    
+    string as_node_name = "asCDL",
+    string as_category = "color",
+    
+    string as_max_class_id = "1669361528 687027596", 
+    string as_max_plugin_type = "texture",
+    
+    int as_maya_type_id = 0x00127a07,
+    string as_maya_classification = "drawdb/shader:rendernode/appleseed/utility"
+]]
+(
+    color in_color = color(0)
+    [[
+        string as_maya_attribute_name = "color",
+        string as_maya_attribute_short_name = "c",
+        string label = "Input Color",
+        string page = "Color",
+        string help = "Input color.",
+        int divider = 1
+    ]],
+    float in_slope = 1.0
+    [[
+        string as_maya_attribute_name = "slope",
+        string as_maya_attribute_short_name = "slo",
+        float min = 0.0,
+        float max = 10000.0,
+        string label = "Slope",
+        string page = "Correction",
+        string help = "Slope (Adjusts the highlights).",
+    ]],
+    float in_offset = 0.0
+    [[
+        string as_maya_attribute_name = "offset",
+        string as_maya_attribute_short_name = "ofs",
+        float min = -1.0,
+        float max = 1.0,
+        string label = "Offset",
+        string page = "Correction",
+        string help = "Constant offset (Adjusts the overall brightness).",
+    ]],
+    float in_power = 1.0
+    [[
+        string as_maya_attribute_name = "power",
+        string as_maya_attribute_short_name = "pow",
+        float min = 0.0,
+        float max = 10000.0,
+        string label = "Power",
+        string page = "Correction",
+        string help = "Over-all exponent (Adjusts the midtones).",
+    ]],
+    float in_saturation = 1.0
+    [[
+        string as_maya_attribute_name = "saturation",
+        string as_maya_attribute_short_name = "sat",
+        float min = 0.0,
+        float max = 10000.0,
+        string label = "Saturation",
+        string page = "Output",
+        string help = "Overall Saturation",
+    ]],
+    output color out_outColor = color(0)
+    [[
+        string as_maya_attribute_name = "outColor",
+        string as_maya_attribute_short_name = "oc",
+        string label = "Output Color"
+    ]]
+)
+{
+    color shifted_color = in_color * in_slope + in_offset;
+    color transformed_color  = pow(shifted_color, in_power);
+    float luma = 0.2126 * transformed_color[0] + 0.7152 * transformed_color[1] + 0.0722 * transformed_color[2];
+    color unclamped = luma + in_saturation * (transformed_color - luma);
+    out_outColor = clamp(unclamped, 0, 1);
```

Comment:
@Mango-3 this will throw away the information from HDRs if the user wants to change the slope of the HDR image (debatable but it was a feature request). So the ideal is to add an *integer as_clamp_output* parameter, widget type *checkBox* so that the user can choose wether to clamp or not, and apply the clamp here according to his/her choice.